src/gpu_constructor.cpp

```diff
@@ -0,0 +1,129 @@
+  *f = new Fluxes(_mixture, _eqSystem, _transport, _num_equation, _dim, axisym);
+}
+
+__global__ void instantiateDeviceRiemann(int _num_equation, GasMixture *_mixture, Equations _eqSystem,
+                                         Fluxes *_fluxClass, bool _useRoe, bool axisym, RiemannSolver **r) {
+  *r = new RiemannSolver(_num_equation, _mixture, _eqSystem, _fluxClass, _useRoe, axisym);
+}
+
+__global__ void freeDeviceMixture(GasMixture *mix) { delete mix; }
+__global__ void freeDeviceTransport(TransportProperties *trans) { delete trans; }
+__global__ void freeDeviceFluxes(Fluxes *f) { delete f; }
+__global__ void freeDeviceRiemann(RiemannSolver *r) { delete r; }
+
+#elif defined(_HIP_)
+
+// __global__ void instantiateDeviceMixture(const WorkingFluid f, const Equations eq_sys,
+//                                          const double viscosity_multiplier, const double bulk_viscosity, int _dim,
+//                                          int nvel, void *mix) {
+//  mix = new (mix) DryAir(inputs, _dim, nvel);
+//}
+__global__ void instantiateDeviceDryAir(const DryAirInput inputs, int _dim, int nvel, void *mix) {
+  mix = new (mix) DryAir(inputs, _dim, nvel);
+}
+
+__global__ void instantiateDeviceDryAirTransport(GasMixture *mixture, const double viscosity_multiplier,
+                                                 const double bulk_viscosity, void *transport) {
+  transport = new (transport) DryAirTransport(mixture, viscosity_multiplier, bulk_viscosity);
+}
+
+__global__ void instantiateDevicePerfectMixture(const PerfectMixtureInput inputs, int _dim, int nvel, void *mix) {
+  mix = new (mix) PerfectMixture(inputs, _dim, nvel);
+}
+
+__global__ void instantiateDeviceFluxes(GasMixture *_mixture, Equations _eqSystem, TransportProperties *_transport,
+                                        const int _num_equation, const int _dim, bool axisym, void *f) {
+  f = new (f) Fluxes(_mixture, _eqSystem, _transport, _num_equation, _dim, axisym);
+}
+
+__global__ void instantiateDeviceRiemann(int _num_equation, GasMixture *_mixture, Equations _eqSystem,
+                                         Fluxes *_fluxClass, bool _useRoe, bool axisym, void *r) {
+  r = new (r) RiemannSolver(_num_equation, _mixture, _eqSystem, _fluxClass, _useRoe, axisym);
+}
+
+__global__ void freeDeviceMixture(GasMixture *mix) {
+  mix->~GasMixture();
+}  // explicit destructor call b/c placement new above
+__global__ void freeDeviceTransport(TransportProperties *transport) { transport->~TransportProperties(); }
+__global__ void freeDeviceFluxes(Fluxes *f) { f->~Fluxes(); }
+__global__ void freeDeviceRiemann(RiemannSolver *r) { r->~RiemannSolver(); }
+
+#endif  // defined(_CUDA_)
+
+// NOTE(kevin): Do not use this. For some unknown reason, this wrapper causes a memory issue, at a random place far
+// after this instantiation.
+void assignMixture(const DryAirInput inputs, const int dim, const int nvel, GasMixture *dMixture) {
```

Comment:
Based on the comment, I assume this isn't being used anywhere, so this doesn't matter, but don't you need to pass `GasMixture **dMixture` for this to work as expected?  Maybe that is the source of the issue.